### ISSUE TITLE
Enable columnar temp tables.

### ISSUE THREAD

**Issue by jeff-davis** [citusdata/citus#4496]

**Comment by jeff-davis** [comment]
@marcocitus found that trivially enabling temporary tables lead to a crash.

SQL:
```
begin; 
create temp table tmp (x int, y int) using columnar on commit drop ;
insert into tmp values (1,20);
commit;
select 1;
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
Time: 171.939 ms
```

Backtrace:
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007f5b389f0859 in __GI_abort () at abort.c:79
#2  0x000055ee34bdade6 in ExceptionalCondition (conditionName=conditionName@entry=0x55ee34d0ee70 "afterTriggers.query_stack == NULL", 
    errorType=errorType@entry=0x55ee34c34028 "FailedAssertion", fileName=fileName@entry=0x55ee34cfdd53 "trigger.c", lineNumber=lineNumber@entry=4408) at assert.c:67
#3  0x000055ee348dacd7 in AfterTriggerBeginXact () at trigger.c:4408
#4  0x000055ee347b4d5b in StartTransaction () at xact.c:2028
#5  0x000055ee347b67b8 in StartTransactionCommand () at xact.c:2828
#6  0x000055ee34aa3aef in start_xact_command () at postgres.c:2639
#7  0x000055ee34aa60e2 in exec_simple_query (query_string=query_string@entry=0x55ee3555fb20 "select 1;") at postgres.c:1017
#8  0x000055ee34aa85f4 in PostgresMain (argc=<optimized out>, argv=argv@entry=0x55ee35600160, dbname=<optimized out>, username=<optimized out>) at postgres.c:4329
#9  0x000055ee34a11c13 in BackendRun (port=port@entry=0x55ee355f8030) at postmaster.c:4526
#10 0x000055ee34a14e7e in BackendStartup (port=port@entry=0x55ee355f8030) at postmaster.c:4210
#11 0x000055ee34a150c7 in ServerLoop () at postmaster.c:1739
#12 0x000055ee34a16606 in PostmasterMain (argc=<optimized out>, argv=<optimized out>) at postmaster.c:1412
#13 0x000055ee3495b7bf in main (argc=3, argv=0x55ee35558b20) at main.c:210
```

It appears to be an interaction between the commit-time flush of columnar data, and triggers on the metadata tables.

**Comment by onurctirtir** [comment]
Closed by #4766
